### PR TITLE
[GAWB-4008] Bond automation test

### PR DESCRIPTION
**Pull request by marctalbott**
Add automation test to check that fence linking works from profile page

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/ProfileSpec.scala [hunk 1]
```diff
@@ -27,5 +28,18 @@ class ProfileSpec extends FreeSpec with WebBrowserSpec with UserFixtures with Ma
 
       }
     }
+
+    "should link with fence" in withWebDriver { implicit driver =>
+      val user = UserPool.chooseStudent
+      withSignIn(user) { _ =>
+        val profilePage = new ProfilePage().open
+
+        profilePage.linkFence
```
Comment (gpolumbo-broad) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/ProfileSpec.scala]: I think this should differentiate between the different providers.  So I think that means we would need to generate the `data-test-id` to have the provider name included, and then your tests should do the same.

**Review by gpolumbo-broad — CHANGES_REQUESTED** [review]

**Review by gpolumbo-broad — APPROVED** [review]

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/ProfilePage.scala [hunk 2]
```diff
@@ -29,4 +29,14 @@ class ProfilePage(implicit webDriver: WebDriver) extends BaseFireCloudPage
     }
     readText(proxyGroupEmailQuery)
   }
+
+  def awaitProvidersReady: Unit = {
+    await notVisible (cssSelector("[data-test-id=spinner]"), 60)
+  }
+
+  def linkProvider(provider: String): Unit = {
```
Comment (gpolumbo-broad) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/ProfilePage.scala]: consider renaming to something more verb like.  Maybe `clickProviderLink` or something

File: src/cljs/main/broadfcui/page/profile.cljs [hunk 3]
```diff
@@ -38,9 +38,9 @@
 (defn get-url-search-param [param-name]
   (.get (js/URLSearchParams. js/window.location.search) param-name))
 
-(defn auth-url-link [href text]
+(defn auth-url-link [href text provider]
   (if href
-    (links/create-external {:href href :target "_self"} text)
+    (links/create-external {:href href :target "_self" :data-test-id provider :id "provider-link"} text)
```
Comment (gpolumbo-broad) [inline src/cljs/main/broadfcui/page/profile.cljs]: IIRC, the `data-test-id` attributes only get rendered on testing environments?  Not sure if that's something you need to add logic for here or if the UI code knows when to ignore these attributes based on whatever environment it's running on.  Maybe @asingh7115 knows the answer?  Or maybe ask a UI person.

Comment (gpolumbo-broad) [inline src/cljs/main/broadfcui/page/profile.cljs]: also, `:id "provider-link` - `id` attributes should be unique on a page.  I think you can change `:id` to `:class` here (and update your test code too) and you should get the same effect.

**Review by gpolumbo-broad — CHANGES_REQUESTED** [review]

File: src/cljs/main/broadfcui/page/profile.cljs (hunk 3 above)
Comment (marctalbott) [reply src/cljs/main/broadfcui/page/profile.cljs]: Just asked her, it shows up on all environments, is that okay?

**Review by dvoet — APPROVED** [review]

**Review by gpolumbo-broad — APPROVED** [review]

**Pull request merged**